### ISSUE TITLE
Don't extend from mapboxgl.Control

### ISSUE THREAD

**Issue by tmcw** [mapbox/mapbox-gl-directions#85]
In index.js:

```
if (window.mapboxgl) {
  mapboxgl.Directions = Directions;
} else if (typeof module !== 'undefined') {
  module.exports = Directions;
}
```

If `window.mapboxgl` is undefined, then this will go require src/directions.js anyway, which has

```js
export default class Directions extends mapboxgl.Control {
```

So there's no real route for the module to work if `window.mapboxgl` is undefined.

**Comment by tmcw** [comment]
Also mapbox-gl-js references the map under this._map https://github.com/mapbox/mapbox-gl-js/blob/master/js/ui/control/control.js#L24 and we save it as this.map https://github.com/mapbox/mapbox-gl-directions/blob/master/src/directions.js#L35 so the inherited .onRemove method will fail.

**Comment by tmcw** [comment]
Fixed in 4fe0b626ab83d05c1e2a9efd9e49463423a6ef80
